packages/components/src/Modal/Modal.rebuilt.test.tsx

```diff
@@ -76,41 +76,153 @@ describe("Composable Modal", () => {
       "secondary destructive",
     );
   });
-  it('modal contains aria role of "dialog"', async () => {
+
+  it("should render the modal header with custom content", () => {
     render(
-      <Modal.Provider open={true}>
+      <Modal.Provider open={true} modalLabelledBy="custom-header">
         <Modal.Content>
-          <Modal.Header title="Modal Title" />
+          <Modal.Header>
+            <span id="custom-header">Custom Header Content</span>
+          </Modal.Header>
         </Modal.Content>
       </Modal.Provider>,
     );
-    expect(await screen.findByRole("dialog")).toBeInTheDocument();
+    expect(screen.getByRole("dialog")).toHaveAttribute(
+      "aria-labelledby",
+      "custom-header",
+    );
+    expect(screen.getByRole("dialog")).toHaveTextContent(
+      "Custom Header Content",
+    );
   });
 
-  it("names the dialog from title via aria-labelledby when ariaLabel is not provided", async () => {
-    render(
-      <Modal.Provider open={true}>
-        <Modal.Content>
-          <Modal.Header title="Billing Settings" />
-        </Modal.Content>
-      </Modal.Provider>,
-    );
+  describe("accessibility", () => {
+    it('modal contains aria role of "dialog"', async () => {
+      render(
+        <Modal.Provider open={true}>
+          <Modal.Content>
+            <Modal.Header title="Modal Title" />
+          </Modal.Content>
+        </Modal.Provider>,
+      );
+      expect(await screen.findByRole("dialog")).toBeInTheDocument();
+    });
 
-    expect(
-      await screen.findByRole("dialog", { name: "Billing Settings" }),
-    ).toBeInTheDocument();
-  });
+    describe("dialog is named from title prop", () => {
+      it("when ariaLabel is not provided", async () => {
```

Comment:
nitpick: the describe and it should be flipped

"when" and other qualifiers are setting the context. "it should..." is the test title, that way it reads as a natural sentence and serves as a layer of free documentation